### PR TITLE
[Debug UI] Migrate NetworkInfoView to new project

### PR DESCRIPTION
**Pull request by robin-near**
See README.md for how to run this locally as well as other information about the new project.

image
image

@saketh-are Added you since you were working on the tier1 network info page. (I wrote this code in December so actually it may be outdated - e.g. the network_info.js you extracted means the js here should be a separate ts file)

### CODE REVIEW

File: tools/debug-ui/src/NetworkInfoView.tsx [hunk 1]
```diff
@@ -148,8 +195,183 @@ export const NetworkInfoView = ({ addr }: NetworkInfoViewProps) => {
...
+    }
+    result += seconds + " s"
+    return result;
+}
+
+function PeerStorageView(addr: string): ReactElement {
+    const [showPeerStorage, setShowPeerStorage] = useState(false);
+
+    const { data: peerStore, error, isLoading } =
+        useQuery(['peer_store', addr], () => fetchPeerStore(addr));
```
Comment (mm-near) [inline tools/debug-ui/src/NetworkInfoView.tsx:311]: BTW - this data is quite large (that's why I've put it behind a button on the debug page).

Is useQuery lazy? or does it start fetching immediately?

File: tools/debug-ui/src/NetworkInfoView.tsx [hunk 2]
```diff
@@ -148,8 +195,183 @@ export const NetworkInfoView = ({ addr }: NetworkInfoViewProps) => {
...
+                    <th>Account ID</th>
+                    <th>Last ping</th>
+                    <th>Height</th>
+                    <th>Last Block Hash</th>
+                    <th>Tracked Shards</th>
+                    <th>Archival</th>
+                    <th>Connection type</th>
+                    <th>First connection</th>
+                    <th>Traffic (last minute)</th>
+                    <th>Route to validators</th>
```
Comment (mm-near) [inline tools/debug-ui/src/NetworkInfoView.tsx:258]: now we also show the connection nonce

Comment (robin-near) [reply tools/debug-ui/src/NetworkInfoView.tsx]: Thanks, added.

**Comment by robin-near** [comment]
Friendly ping

File: tools/debug-ui/src/CurrentPeersView.tsx [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    if (!fullStatus || !epochInfo) {
+        return <div className="network-info-view">
+            <div className="error">No Data</div>
+        </div>;
+    }
+
+    const detailedDebugStatus = fullStatus.detailed_debug_status!;
+
+    return <div className="current-peers-view">
+        <p>Current Sync Status: {detailedDebugStatus.sync_status}</p>
```
Comment (saketh-are) [inline tools/debug-ui/src/CurrentPeersView.tsx:195]: This section could benefit from some formatting to improve readability. Maybe we could have the data descriptions in a different font or weight than the actual data? For example **Number of peers**: 3 / 40. Screenshot of current appearance:

image

File: tools/debug-ui/src/CurrentPeersView.tsx [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    }
+    if (!fullStatus || !epochInfo) {
+        return <div className="network-info-view">
+            <div className="error">No Data</div>
+        </div>;
+    }
+
+    const detailedDebugStatus = fullStatus.detailed_debug_status!;
+
+    return <div className="current-peers-view">
```
Comment (saketh-are) [inline tools/debug-ui/src/CurrentPeersView.tsx:194]: We also display the node's PeerId at the top now: https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/network_info.html#L227

File: tools/debug-ui/src/CurrentPeersView.tsx [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                    <th>Nonce</th>
+                    <th>First connection</th>
+                    <th>Traffic (last minute)</th>
+                    <th>Route to validators</th>
+                </tr>
+            </thead>
+            <tbody>
+                {peers.map(({ peer, validator, routedValidator, statusClassName }) => {
+                    return <tr key={peer.peer_id}>
+                        <td>{addDebugPortLink(peer.addr)}</td>
```
Comment (saketh-are) [inline tools/debug-ui/src/CurrentPeersView.tsx:263]: This is linking to the old-style debug page

**Review by saketh-are — APPROVED** [review]
@robin-near this looks good to me!

We also now display "recent outbound connections" on this page: https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/network_info.html#L320. It works very similarly to the detailed peer storage display and it would be nice to have a ConnectionStoreView showing that information. You could add it here or I'd be happy to send a separate PR adding it.

**Review by mzhangmzz — APPROVED** [review]

**Pull request merged**